README.md

```diff
@@ -67,6 +67,53 @@ This Solidity project uses Foundry as a toolkit. If you don't have Foundry insta
     forge snapshot
     ```
 
+### Generating Code Coverage Report
+
+1. **Generate the `lcov` File:**
+    To generate the coverage data in `lcov` format, run the following command:
+    ```sh
+    forge coverage --report lcov
+    ```
+
+2. **Install `lcov`:**
+    If you don't have `lcov` installed, you can install it using your package manager. Here are the instructions for different operating systems:
+
+    - **macOS (using Homebrew):**
+      ```sh
+      brew install lcov
+      ```
+
+    - **Linux (using apt for Debian/Ubuntu):**
+      ```sh
+      sudo apt-get update
+      sudo apt-get install lcov
+      ```
+
+    - **Linux (using yum for CentOS/Fedora):**
+      ```sh
+      sudo yum install lcov
+      ```
+
+3. **Filter the Coverage Data:**
+    To ensure that only files from the `src` directory are included in the report, filter the coverage data:
+    ```sh
+    lcov --remove lcov.info 'test/*' -o lcov_filtered.info --ignore-errors inconsistent
+    ```
+
+4. **Generate the HTML Report:**
+    To create an HTML report from the filtered `lcov` data, use:
+    ```sh
+    genhtml lcov_filtered.info --output-directory coverage-report --ignore-errors inconsistent,corrupt
```

Comment:
you want to instead address the specific error only::

```
genhtml lcov_filtered.info --output-directory coverage-report --rc derive_function_end_line=0
```

see the warning in the forge docs for explanation https://book.getfoundry.sh/reference/forge/forge-coverage#description